### DISCUSSION TITLE
How to use brew to update installed packages?

### DISCUSSION THREAD

**Discussion by DataSailor** [Homebrew/discussions#5192] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.11
ORIGIN: https://github.com/Homebrew/brew
HEAD: d7d4c8266210c024b93a450a7d357cec0b46a1bb
Last commit: 7 days ago
Core tap JSON: 11 Mar 04:18 UTC
Core cask tap JSON: 11 Mar 04:18 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 16-core 64-bit arm_palma
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.3.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.3
Rosetta 2: false
```

### Output of `brew doctor`

```shell
brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed casks are deprecated or disabled.
You should find replacements for the following casks:
  cakebrew
```

### Description of issue

I recently started using Mac and tried to set up my development environment with Homebrew.
But I found that I couldn't update the installed packages normally.

According to the information I found, the commands to update software packages are `brew update` and `brew upgrade`.
But these commands don't seem to be able to update my software packages.
<img width="1045" alt="image" src="https://github.com/Homebrew/discussions/assets/39119487/13be7a31-60fe-4f9c-8027-aa8fd7fb58be">

In addition, after I updated miniconda using `brew install`, I found that the virtual environment I had previously created disappeared.

before update:
<img width="619" alt="image" src="https://github.com/Homebrew/discussions/assets/39119487/ddead3b2-177c-4702-9c8e-fa09bc5d6e0b">

after update:
<img width="550" alt="image" src="https://github.com/Homebrew/discussions/assets/39119487/d04ce2a9-0d36-4bdd-9f61-321e77be5be0">

So, how to safely automatically update the installed software packages?

Thank you for help.

**Comment by ZhongRuoyu** [comment, marked as answer]
`miniconda` can update itself (through `conda update -n base conda`), so Homebrew does not attempt to upgrade it by default. Please see ["Why aren’t some apps included during `brew upgrade`?"](https://docs.brew.sh/FAQ#why-arent-some-apps-included-during-brew-upgrade) for more details.

As for why your environment disappeared: when you upgrade `miniconda` through `brew upgrade`, it is effectively reinstalled, and currently it does not seem possible to preserve environments through re-installation. See conda/conda#8074.

**Reply by DataSailor** [reply]
Thank you for your answer.

Another question is, how to check which software has the `auto_updates true` flag?

**Reply by ZhongRuoyu** [reply]
You can check if the first line of `brew info --cask` output ends with `(auto_updates)`.

For example, `miniconda` auto updates:

```console
$ brew info --cask miniconda
==> miniconda: py312_24.1.2-0 (auto_updates)
[output truncated]
```

While `balenaetcher` does not:

```console
$ brew info --cask balenaetcher
==> balenaetcher: 1.18.11
[output truncated]
```
